### PR TITLE
MAT-9745: add Admin APIs (create, update, delete) for CodeSystem

### PR DESCRIPTION
**Pull request by etan-sb**
## Terminology Service PR

Jira Ticket: [MAT-9745](https://jira.cms.gov/browse/MAT-9745)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/controller/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    return ResponseEntity.status(HttpStatus.CREATED)
+        .body(fhirTerminologyService.createCodeSystem(codeSystem));
+  }
+
+  @PutMapping(
+      path = "/{id}",
+      produces = MediaType.APPLICATION_JSON_VALUE,
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<CodeSystem> updateCodeSystem(
+      Principal principal, @PathVariable String id, @Valid @RequestBody CodeSystem codeSystem) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/controller/AdminController.java:62]: Might be an easier to lookup existing code systems by their OID/URL and vsac/fhir version since we'll have to do some digging to find the document ID.

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java [hunk 2]
```diff
@@ -469,6 +470,51 @@ private void updateOrInsertAllCodeSystems(List<CodeSystem> codeSystemList) {
...
+  public CodeSystem updateCodeSystem(String id, CodeSystem codeSystem) {
+    CodeSystem existing =
+        codeSystemRepository
+            .findById(id)
+            .orElseThrow(
+                () -> new CodeSystemNotFoundException("CodeSystem not found for id: " + id));
+    existing.setFullUrl(codeSystem.getFullUrl());
+    existing.setTitle(codeSystem.getTitle());
+    existing.setName(codeSystem.getName());
+    existing.setVersion(codeSystem.getVersion());
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: It's a bit more code, but setting the individual versions reduces the chances of accidentally wiping one out if the caller doesn't supply it.

Which, now that I think about it, is true for all of the setters.

Comment (etan-sb) [reply src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: Updated

File: src/main/java/gov/cms/madie/terminology/controller/AdminController.java (hunk 1 above)
Comment (etan-sb) [reply src/main/java/gov/cms/madie/terminology/controller/AdminController.java:62]: After discussion with Joe, keeping document ID for lookup

File: src/main/java/gov/cms/madie/terminology/controller/AdminController.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  @PutMapping(
+      path = "/code-system/{id}",
+      produces = MediaType.APPLICATION_JSON_VALUE,
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  @PreAuthorize("hasRole('MADIE-ADMIN')")
+  public ResponseEntity<CodeSystem> updateCodeSystem(
+      Principal principal, @PathVariable String id, @Valid @RequestBody CodeSystem codeSystem) {
+    log.info(
+        "Admin user [{}] is updating code system with id: [{}]"
+            + " name: [{}] oid: [{}] fullUrl: [{}] version: [{}]",
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/controller/AdminController.java:65]: Does the version value print nicely or are you seeing the object address?

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/main/java/gov/cms/madie/terminology/controller/AdminController.java (hunk 3 above)
Comment (etan-sb) [reply src/main/java/gov/cms/madie/terminology/controller/AdminController.java:65]: It would look something like this:

`Admin user [x] is updating code system with id: [69bb1ce87865766fd881e749] name: [TestCode] oid: [urn:oid:9.9.999.999999.9.9] fullUrl: [http://terminology.hl7.org/CodeSystem/v3-TestCode] version: [CodeSystem.Version(fhirVersion=2.0.0, vsacVersion=1.0.0)]`

**Review by jkotanchik-SB — APPROVED** [review]
